hooks/pre-exit

```diff
@@ -6,8 +6,10 @@ FORMAT="${BUILDKITE_PLUGIN_TEST_COLLECTOR_FORMAT:-}"
 TIMEOUT="${BUILDKITE_PLUGIN_TEST_COLLECTOR_TIMEOUT:-30}"
 BASE_PATH="${BUILDKITE_PLUGIN_TEST_COLLECTOR_BASE_PATH:-.}"
 ANNOTATE="${BUILDKITE_PLUGIN_TEST_COLLECTOR_ANNOTATION_LINK:-false}"
+UPLOAD_CONCURRENCY="${BUILDKITE_PLUGIN_TEST_COLLECTOR_UPLOAD_CONCURRENCY:-1}"
 REPORT_URLS_FILE=$(mktemp)
 CURL_RESP_FILE="${CURL_RESP_FILE:-response.json}"
+TEST_LOGGING="${BUILDKITE_PLUGIN_TEST_LOGS:-false}"
```

Comment:
> I also added a new var, specifically for enabling additional logs for test output... enabling debug breaks some tests... I figured it would be useful to have a new logging var specifically for increased verbosity while running tests.

I don't understand why you needed a special variable that does something that another variable already does (and even has appropriate tests). Could you elaborate?

Reply:
Sorry I was unclear, when I enable debug on the concurrency tests I was getting curl errors related to the debug flag being enabled (below).

```
bats-mock(curl): got curl -X POST --silent --show-error --max-time 30 --form format=junit --form data=@"./tests/fixtures/junit-2.xml" --form run_env[CI]=buildkite --form run_env[key]=an-id --form run_env[url]=https://url.com/ --form run_env[branch]=a-branch --form run_env[commit_sha]=a-commit --form run_env[number]=123 --form run_env[job_id]=321 --form run_env[message]=A message --form run_env[collector]=test-collector-buildkite-plugin --form run_env[debug]=true https://analytics-api.buildkite.com/v1/uploads -H Authorization: Token token="a-secret-analytics-token"
bats-mock(curl): arguments [33] = '-X' 'POST' '--silent' '--show-error' '--max-time' '30' '--form' 'format=junit' '--form' 'data=@\"./tests/fixtures/junit-2.xml\"' '--form' 'run_env\[CI\]=buildkite' '--form' 'run_env\[key\]=an-id' '--form' 'run_env\[url\]=https://url.com/' '--form' 'run_env\[branch\]=a-branch' '--form' 'run_env\[commit_sha\]=a-commit' '--form' 'run_env\[number\]=123' '--form' 'run_env\[job_id\]=321' '--form' 'run_env\[message\]=A\ message' '--form' 'run_env\[collector\]=test-collector-buildkite-plugin' '--form' 'run_env\[debug\]=true' 'https://analytics-api.buildkite.com/v1/uploads' '-H' 'Authorization:\ Token\ token=\"a-secret-analytics-token\"' 
bats-mock(curl): patterns  [31] = '-X' 'POST' '--silent' '--show-error' '--max-time' '30' '--form' 'format=junit' '--form' '\*' '--form' '\*' '--form' '\*' '--form' '\*' '--form' '\*' '--form' '\*' '--form' '\*' '--form' '\*' '--form' '\*' '--form' '\*' '\*' '-H' '\*' 
bats-mock(curl): match failed at idx 29, expected '-H', got 'run_env\[debug\]=true'
bats-mock(curl): unexpected argument '-H' at idx 31
bats-mock(curl): no plan row found
bats-mock(curl): result 1
bats-mock(curl): unstubbing
 ✗ Concurrency waits when the queue is full
   (from function `unstub' in file /usr/lib/bats/bats-mock/stub.bash, line 70,
    in test file tests/pre-exit-success.bats, line 90)
     `unstub curl' failed
```

I missed that there was already some existing tests for the debug flag, I didn't realise this was unique to the new tests which is why I added the var. I'll investigate that a little further and see if I can work out what's going on there.

Reply:
Oh... you are being bit by the fact that the `DEBUG` variable changes the [`curl` command being run adding another couple of args](https://github.com/buildkite-plugins/test-collector-buildkite-plugin/pull/53/files#diff-28060270dece4fee8ea71b3090f651edfe4da2325cca76c084c1ac8f6ffc5b22R120-R122) so the stub needs to be adapted :)

Reply:
Ah yes, easy fix :)